daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala

```diff
@@ -46,6 +73,17 @@ class DarReader[A](
   private def close(is: InputStream): Try[Unit] = Try(is.close())
 }
 
+case class InvalidLegacyDar(dalfNames: List[String])
+    extends RuntimeException(
+      s"Invalid Legacy DAR. Legacy DAR can contain: one main DALF and an optional *-prim* DALF. " +
+        s"This DAR contains: [${dalfNames.mkString(", "): String}]")
+
+case class InvalidDar(cause1: Throwable, cause2: Throwable)
```

Comment:
I wouldn't mention the legacy mode at all. If both "modes" fail to read the DAR file, we should just fail with the error message of the currently expected format (aka DAR with manifest). For most users the legacy mode should never come into action.

Reply:
i'm happy for @leo-da to adjust the error message, i'll merge this in the meantime.